### PR TITLE
SynthConfig as non-Tensors

### PR DESCRIPTION
**Pull request by turian**
https://github.com/turian/torchsynth/pull/267https://github.com/turian/torchsynth/pull/267 should be merged first for a smaller diff

I am trying to simplify the synth configuration, and it will be a sequence of linear PRs. Here is one that turns most of the configs back into Python native values.

@jorshi can you profile quickly to make sure this is okay before we merge?

### CODE REVIEW

**Comment by turian** [comment]
@jorshi can you profile this? Should be fine but let's double check

File: torchsynth/config.py [hunk 1]
```diff
@@ -17,42 +10,38 @@ class SynthConfig:
...
         self,
-        batch_size: T,
-        sample_rate: Optional[T] = tensor(DEFAULT_SAMPLE_RATE),
-        buffer_size: Optional[T] = tensor(DEFAULT_BUFFER_SIZE),
-        control_rate: Optional[T] = tensor(DEFAULT_CONTROL_RATE),
+        batch_size: int,
+        sample_rate: Optional[int] = 44100,
+        buffer_size_seconds: Optional[float] = 4.0,
+        control_rate: Optional[int] = 441,
+        debug: bool = "TORCHSYNTH_DEBUG" in os.environ,
```
[the remark this answers is not in the document]
Comment (turian) [reply torchsynth/config.py:21]: Okay so check the next pr

https://github.com/turian/torchsynth/pull/269/files#diff-fb6a686182f16eb54af3c628f38593f347f68aba31de903983023c560288d7a1

basically in pytest.ini it sets this variable for you and you never have to touch this env variable

If you do feel strongly that the value should matter then okay. I'm fine with it just existing as an environment variable. But I can fix if you feel this is important

File: torchsynth/config.py [hunk 2]
```diff
@@ -17,42 +10,38 @@ class SynthConfig:
...
         # Buffer size for control signals -- this is calculated to have the
         # same duration in seconds as that buffer size for the audio rate
         # signals. Rounded to the nearest integer number of samples.
-        self.control_buffer_size = (
-            torch.round((buffer_size / sample_rate * control_rate))
-            .clone()
-            .detach()
-            .int()
+        self.control_buffer_size = torch.tensor(
+            int(torch.round((self.buffer_size / sample_rate * control_rate)))
```
[the remark this answers is not in the document]
Comment (turian) [reply torchsynth/config.py:44]: adding back to resolve

File: torchsynth/profile.py [hunk 3]
```diff
@@ -15,27 +15,24 @@
...
-# Turn off torchsynth debug mode before importing anything else
-# TODO: https://github.com/turian/torchsynth/issues/259
-import torchsynth.oldconfig
-
-torchsynth.oldconfig.DEBUG = False
 
 import torchsynth.synth  # noqa: E402
-from torchsynth.synth import AbstractSynth  # noqa: E402
 from torchsynth.config import SynthConfig  # noqa: E402
+from torchsynth.synth import AbstractSynth  # noqa: E402
```
[the remark this answers is not in the document]
Comment (turian) [reply torchsynth/profile.py]: Fixed.

a) For these thing if there is a warning string let's use that not the cryptic name, but the full string of the warning we're suppressing
b) I don't think profile.py is a package module, it belongs in examples/ or scripts/
Anyway, filed an issue todo that later.

File: torchsynth/config.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jorshi) [reply torchsynth/config.py:21]: Nice, ya, I'm okay with it if we just make note of that in dev docs.

File: torchsynth/profile.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (jorshi) [reply torchsynth/profile.py]: a) agree

b) I also agree -- but afaik there isn't a good way to put this in scripts, have it as a .py and invoke it using torchsynth.profile -- but let's look into it

**Pull request merged**